python/ml4ir/applications/ranking/model/ranking_model.py

```diff
@@ -70,34 +75,28 @@ def predict(
             logging_frequency=logging_frequency,
         )
 
-
-    def run_ttest(self, mean, variance, n, ttest_pvalue_threshold):
-        """
-        Compute the paired t-test statistic and its p-value given mean, standard deviation and sample count
-        Parameters
-        ----------
-        mean: float
-            The mean of the rank differences for the entire dataset
-        variance: float
-            The variance of the rank differences for the entire dataset
-        n: int
-            The number of samples in the entire dataset
-        ttest_pvalue_threshold: float
-            P-value threshold for student t-test
-
-        Returns
-        -------
-        t_test_metrics_dict: Dictionary
-            A dictionary with the t-test metrics recorded.
-        """
-        t_test_stat, pvalue = perform_click_rank_dist_paired_t_test(mean, variance, n)
-        t_test_log_results(t_test_stat, pvalue, ttest_pvalue_threshold, self.logger)
-
-        # Log t-test statistic and p-value
-        t_test_metrics_dict = {'Rank distribution t-test statistic': t_test_stat,
-                               'Rank distribution t-test pvalue': pvalue,
-                               'Rank distribution t-test difference is statistically significant': pvalue < ttest_pvalue_threshold}
-        return t_test_metrics_dict
+    def prepare_eval_config(self, file_io, evaluation_config_path, group_key):
+        eval_dict = {}
+        if file_io:
+            eval_config = file_io.read_yaml(evaluation_config_path)
+            if "group_by" in eval_config["power_analysis"] and eval_config["power_analysis"]["group_by"]:
+                eval_dict["group_by"] = eval_config["power_analysis"]["group_by"].split(',')
+            else:
+                eval_dict["group_by"] = group_key
+            eval_dict["metrics"] = eval_config["power_analysis"]["metrics"].replace(" ","").split(',')
+            eval_dict["variance_list"] = []
+            for m in eval_dict["metrics"]:
+                eval_dict["variance_list"].append("old_" + m)
+                eval_dict["variance_list"].append("new_" + m)
+            eval_dict["power"] = float(eval_config["power_analysis"]["power"])
+            eval_dict["pvalue"] = float(eval_config["power_analysis"]["pvalue"])
+        else:
+            eval_dict["group_by"] = group_key
+            eval_dict["metrics"] = []
+            eval_dict["variance_list"] = []
+            eval_dict["power"] = None
+            eval_dict["pvalue"] = RankingConstants.TTEST_PVALUE_THRESHOLD
+        return eval_dict
```

Comment:
@mohazahran As mentioned elsewhere, initialize this in the pipeline itself. And create a new class call `EvalConfig` as this looks like a lot of customization.